CommonData/column.cpp

```diff
@@ -1661,12 +1666,29 @@ bool Column::setStringValueToRowIfItFits(size_t row, const std::string & value,
                                 ColumnUtils::convertValueToDoubleForImport(value,				dbl))
 							typeChanged = changeType(_preEditType) == columnTypeChangeResult::changed; //Just try it
 					}
+					else if(_type == columnType::nominalText)
+					{
+						// If all the values are now integers, change the column to nominal
+						bool onlyIntegers = true;
+						for(size_t i=0; i<_labels.size(); i++)
+						{
+							if (!_labels[i]->originalValue().isInt())
+							{
+								onlyIntegers = false;
+								break;
+							}
+						}
+
+						if (onlyIntegers)
+							_changeColumnToNominalOrOrdinal(columnType::nominal);
```

Comment:
```suggestion
							_changeColumnToNominalOrOrdinal(_prevEditType);
```
And this?

Btw, I didnt test this but it seems like better behaviour